src/utils/query/createExtensionStoragePersister.ts

```diff
@@ -0,0 +1,36 @@
+import type { PersistedClient, Persister } from "@tanstack/query-persist-client-core";
+import { ExtensionStorage } from "~utils/storage";
+
+/**
+ * Creates a persister that uses ExtensionStorage for React Query cache persistence
+ * @param options Configuration options
+ * @param options.cacheKey Unique key to use for storing the cache in ExtensionStorage
+ * @returns A persister object that can be used with persistQueryClient
+ */
+export function createExtensionStoragePersister({ cacheKey }: { cacheKey: string }): Persister {
+  return {
+    persistClient: async (persistClient: PersistedClient) => {
+      try {
+        await ExtensionStorage.set(cacheKey, JSON.stringify(persistClient));
```

Comment:
You don't need to use `JSON.stringify` here or anywhere with ExtensionStorage as it will be handled internally.